docs/src/cli/deploy-a-program.md

```diff
@@ -0,0 +1,217 @@
+---
+title: Deploy a Program
+---
+
+Developers can deploy on-chain [programs](terminology.md#program) (often called
+smart contracts elsewhere) with the Solana tools.
+
+To learn about developing and executing programs on Solana, start with the
+[overview](developing/programming-model/overview.md) and then dig into the
+details of [deployed programs](developing/deployed-programs/overview.md).
+
+To deploy a program, use the Solana tools to interact with the on-chain loader
+to:
+
+- Initialize a program account
+- Upload the program's shared object to the program account's data buffer
+- Verify the uploaded program
+- Finalize the program by marking the program account executable.
+
+Once deployed, anyone can execute the program by sending transactions that
+reference it to the cluster.
+
+## Usage
+
+### Deploy a program
+
+To deploy a program, you will need the location of the program's shared object
+(the program binary .so)
+
+```bash
+solana program deploy <PROGRAM_FILEPATH>
+```
+
+Successful deployment will return the program id of the deployed program, for
+example:
+
+```bash
+Program Id: 3KS2k14CmtnuVv2fvYcvdrNgC94Y11WETBpMUGgXyWZL
+```
+
+Specify the keypair in the deploy command to deploy to a specific program id:
+
+```bash
+solana program deploy --program-id <KEYPAIR_FILEPATH> <PROGRAM_FILEPATH>
+```
+
+If the program id is not specified on the command line the tools will first look
+for a keypair file matching the `<PROGRAM_FILEPATH>`, or internally generate a
+new keypair.
+
+A matching program/keypair file is generated automatically by the program build
+tools and looks like:
```

Comment:
Suggest:

A matching program keypair file is in the same directory as the program's shared object, and named `<PROGRAM_NAME>-keypair.json`. Matching program keypairs are generated automatically by the program build tools: